.gitlab-ci.yml

```diff
@@ -0,0 +1,21 @@
+image: ubuntu:bionic
+
+variables:
+  GRADLE_VERSION: "5.4.1"
+  
+before_script:
+  - apt-get update && apt-get -y upgrade
+  - apt-get -y install openjdk-11-jdk gradle
+  - gradle wrapper --gradle-version=$GRADLE_VERSION
+  - chmod +x ./gradlew
+
+build:
+  script:
+    - java -version
+    - LC_ALL=C.UTF-8 ./gradlew build
```

Comment:
Is the environment variable necessary? If yes, why? If not, can we remove the assignment?

Reply:
It's needed to prevent fatal errors generating JavaDoc due to character set. There may be another, better way of resolving the issue, but I'm pretty sure if I remove this it will be broken. I'll verify.

Reply:
Hmm, why is this not an issue locally or on Travis?

Maybe adding

```
javadoc {
    options.encoding = 'UTF-8'
}
```

to build.gradle could also help?

Reply:
Trying that now...

Reply:
Did it work?

Reply:
It didn't work. (and I got distracted by something else, sorry.)

Can we go ahead and merge this as is, and fix it in a separate PR when we find a better way?

Reply:
Well, I'm not too happy with "shell foo" in build scripts but ok, for now I'm fine with it.

Reply:
Weird, I just reproduced your issue by removing the environment varable assignment. Then I fixed the issue by adding `javadoc.options.encoding = 'UTF-8'` to the core build.gradle, which should be identical to my above suggestion.

Reply:
OK, I'll rebase and see what happens.

Reply:
Yeah, `javadoc.options.encoding = 'UTF-8'` fixes the build failure (don't recall exactly why my previous attempt at setting javadoc options failed.

However, the test output still has character set issues, so maybe we need to add one more setting in `build.gradle`. Also, we probably want to remove/replace the `--info` parameter to the two Gradle commands.

Reply:
> However, the test output still has character set issues

I was seeing `???` in the logging timestamps from the tests when running with Docker and `gitlab-runner` under macOS, but this is no longer happening.

We still might want to figure out how to set UTF-8 as the default character set at the system or Gradle level rather than task by task, but we can worry about that later if we need to.